cpp/include/cudf/structs/structs_column_device_view.cuh

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/types.hpp>
+
+namespace cudf {
+
+namespace detail {
+
+/**
+ * @brief Given a column-device-view, an instance of this class provides a
+ * wrapper on this compound column for struct operations.
+ * Analogous to list_column_view.
+ */
+class structs_column_device_view {
```

Comment:
I think this and `list_column_device_view` could've simply inherited from `column_device_view` rather than compose if not for the fact that we want to hide the `data()` member because neither has data of its own. And some other methods which don't make sense for them. But since these are constructed from a `column_device_view`, nothing stops someone from just accessing those from the `column_device_view` itself.

Reply:
How did I forget about private inheritance. That makes so much sense